Argo/Globals/Parser.swift

```diff
@@ -0,0 +1,26 @@
+import Box
+
+public enum Parser<T> {
+  case Success(Box<T>)
+  case TypeMismatch(String)
+  case MissingKey(String)
+
+  public var value: T? {
+    switch self {
+    case let .Success(box): return box.value
+    default: return .None
+    }
+  }
+
+  public static func typeMismatch(expectedType: String, object: JSON) -> Parser<T> {
+    return .TypeMismatch("\(object) is not a \(expectedType)")
+  }
+}
+
+func parseOptional<A>(p: Parser<A>) -> Parser<A?> {
```

Comment:
we decided that a "key missing" is an acceptable error to ignore when the user uses the `<|?` operator because they are saying "This may or may not be there"; however, if it is there and it fails from casting we want that error to carry through because that's something we think the user should address.